jwst/resample/resample.py

```diff
@@ -179,14 +180,16 @@ def __init__(self, input_models, output=None, single=False, blendheaders=True,
         self.blank_output = datamodels.ImageModel(tuple(self.output_wcs.array_shape))
 
         # update meta data and wcs
-        self.blank_output.update(input_models[0])
+        with input_models:
+            example_model = input_models.borrow(0)
+            input_models.shelve(example_model, 0, modify=False)
+        self.blank_output.update(example_model)
```

Comment:
```suggestion
            self.blank_output.update(example_model)
            input_models.shelve(example_model, 0, modify=False)
            del example_model
```
To let `example_model` go out-of-scope.